presto-hive/src/main/java/com/facebook/presto/hive/RecordFileWriter.java

```diff
@@ -161,6 +163,8 @@ public void rollback()
     {
         try {
             recordWriter.close(true);
+            // perform explicit delete of written file here as recordWriter.close() ignore the abort flag.
+            path.getFileSystem(conf).delete(path, false);
```

Comment:
I don't know. When I tested that it seems to behave correctly - the files are deleted - even though delete is issed immediatly after closing file outputstream. But there may be some corner cases.